src/Components/TodoList.tsx

```diff
@@ -1,52 +1,69 @@
+import React from "react";
 import { useState } from "react";
-import EditTodo from "./Dropdown";
 import {
   Box,
-  Checkbox,
+  Button,
+  Dialog,
+  DialogActions,
+  DialogTitle,
   Grid,
   IconButton,
   Paper,
   Stack,
+  TextField,
   Typography,
 } from "@mui/material";
 import DeleteIcon from "@mui/icons-material/Delete";
 import EditIcon from "@mui/icons-material/Edit";
-import { styled } from "@mui/material/styles";
 
 interface Props {
-  isCompleted: boolean | undefined;
-  checkBoxClick: (value: boolean) => void;
   todoText: string;
   handleClick: () => void;
   handleUpdateTodo: (value: string) => void;
 }
-const TodoList = ({
-  isCompleted,
-  checkBoxClick,
-  todoText,
-  handleClick,
-  handleUpdateTodo,
-}: Props) => {
-  const label = { inputProps: { "aria-label": "Checkbox demo" } };
+const TodoList = ({ todoText, handleUpdateTodo, handleClick }: Props) => {
   const [editTodo, setEditTodo] = useState(false);
-  const Item = styled(Paper)(({ theme }) => ({
-    backgroundColor: theme.palette.mode === "dark" ? "#1A2027" : "#fff",
-    ...theme.typography.body2,
-    padding: theme.spacing(1),
-    textAlign: "center",
-    color: theme.palette.text.secondary,
-    maxWidth: 400,
-  }));
+  const [open, setOpen] = useState(false);
+  const [editedTodo, setEditedTodo] = useState(todoText);
+
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const handleEdit = () => {
+    setEditTodo(true);
+    setOpen(false); // Close the delete dialog when edit mode is activated
+  };
+
+  const handleSaveEdit = () => {
+    handleUpdateTodo(editedTodo);
+    setEditTodo(false);
+  };
   return (
     <>
+      <Dialog
+        open={open}
```

Comment:
change the variable name `shouldOpenDialog`
```suggestion
        open={shouldOpenDialog}
```